src/main.ts

```diff
@@ -135,54 +164,45 @@ export default class DeepLPlugin extends Plugin {
 		});
 
 		this.addCommand({
-			id: "deepl-translate-selection-from-to",
-			name: "Translate selection: From a language to another",
-			editorCallback: async (editor: Editor) => {
-				if (editor.getSelection() === "") {
-					return;
-				}
-
-				new TranslateModal(
-					app,
-					"From",
-					Object.entries(fromLanguages).map(([code, name]) => ({
-						code,
-						name,
-					})),
-					async (from) => {
-						new TranslateModal(
-							app,
-							"To",
-							Object.entries(toLanguages).map(([code, name]) => ({
-								code,
-								name,
-							})),
-							async (to) => {
-								try {
-									const translation =
-										await this.deeplService.translate(
-											editor.getSelection(),
-											to.code,
-											from.code
-										);
-									editor.replaceSelection(
-										translation[0].text
-									);
-								} catch (error) {
-									if (error instanceof DeepLException) {
-										new Notice(error.message);
-									} else {
-										console.error(error, error.stack);
-										new Notice(
-											"An unknown error occured. See console for details."
-										);
-									}
-								}
-							}
-						).open();
-					}
-				).open();
-			},
+		    id: "deepl-translate-selection-append",
+		    name: "Translate selection: To language and append to selection",
+		    editorCallback: async (editor: Editor) => {
+		        if (editor.getSelection() === "") {
+		            return;
+		        }
+
+		        const selection = editor.getSelection();
+
+		        new TranslateModal(
```

Comment:
I can only specify one language in the settings. What if I want to translate from another language that is not set? That was the reason I added this functionality back then. I would miss this, if there are better solutions, I am very much in favour of them.

The command `deepl-translate-selection-to` use the language from the settings. The original command was explicitly meant to be able to choose a language that does not come from the settings.

Reply:
It is not removing it.. It's only adding a new option without the modal. Not changing the existing modal function.

Reply:
In my GitHub diff view, I see that the `deepl-translate-selection-from-to` command has been removed. More precisely, it has been replaced with `deepl-translate-selection-append`. On the right side of the diff, the command `deepl-translate-selection-from-to` is no longer present. I can no longer see the `from modal` implementation either. But I only looked at it in the GitHub diff, I will check out the code, maybe an error in the diff view.

Reply:
Ok, now that I'm at the computer and can review this.. you're right actually. This is a mistake, and **I appreciate you caught this**! It's also super embarrassing.. 🤦‍♂️

I did not mean to remove this function. This would not be an improvement, and as you say I would miss this function also.